server/handlers/component_handler.go

```diff
@@ -1296,7 +1378,7 @@ func (h *Handler) RegisterMeshmodelComponents(rw http.ResponseWriter, r *http.Re
 func (h *Handler) GetMeshmodelRegistrants(rw http.ResponseWriter, r *http.Request) {
 	rw.Header().Add("Content-Type", "application/json")
 	enc := json.NewEncoder(rw)
-
+	registry.Mutex.Lock()
```

Comment:
@MUzairS15 should i use this to sync the registering and fetching of the data i Don't think it is necessary just a fail safe if someone opens the server if the entites are still being registered can you confirm this 
https://github.com/meshery/meshery/blob/dd67ed6e06ec6070cff8356ca5b3e9b641fa7d58/server/meshmodel/helper.go#L167

Reply:
No not required